lstchain/tools/tests/test_tools.py

```diff
@@ -84,6 +84,66 @@ def test_create_irf_full_enclosure_with_config(temp_dir_observed_files, simulate
     )
 
 
+def test_create_irf_point_like_energy_dependent_cuts(
+    temp_dir_observed_files, simulated_dl2_file
+):
+    """
+    Generating point-like IRF file from a test DL2 files, using
+    energy-dependent cuts
+    """
+    from lstchain.tools.lstchain_create_irf_files import IRFFITSWriter
+
+    irf_file = temp_dir_observed_files / "irf.fits.gz"
+
+    assert (
+        run_tool(
+            IRFFITSWriter(),
+            argv=[
+                f"--input-gamma-dl2={simulated_dl2_file}",
+                f"--input-proton-dl2={simulated_dl2_file}",
+                f"--input-electron-dl2={simulated_dl2_file}",
+                f"--output-irf-file={irf_file}",
+                "--overwrite",
+                "--energy-dependent-gh",
+                "--point-like",
+                "--energy-dependent-theta"
+            ],
+            cwd=temp_dir_observed_files,
+        )
+        == 0
+    )
+
+
+@pytest.mark.private_data
+def test_create_dl3_energy_dependent_cuts(
+    temp_dir_observed_files, observed_dl2_file
+):
+    """
+    Generating an DL3 file from a test DL2 files and test IRF file, using
+    energy dependent cuts. Here the previously created IRF is used.
+    """
+    from lstchain.tools.lstchain_create_dl3_file import DataReductionFITSWriter
+
+    irf_file = temp_dir_observed_files / "irf.fits.gz"
+
+    assert (
+        run_tool(
+            DataReductionFITSWriter(),
+            argv=[
+                f"--input-dl2={observed_dl2_file}",
+                f"--output-dl3-path={temp_dir_observed_files}",
+                f"--input-irf={irf_file}",
+                "--source-name=Crab",
+                "--source-ra=83.633deg",
+                "--source-dec=22.01deg",
+                "--overwrite",
+            ],
+            cwd=temp_dir_observed_files,
+        )
+        == 0
+    )
```

Comment:
Please at some tests for the produced output. E.g. that the file can be read using `Observation.read` from gammapy and that all expected data are there.